src/domains/ecommerce-app/05_functions.tf

```diff
@@ -0,0 +1,128 @@
+resource "azurerm_resource_group" "ecommerce_functions_rg" {
+  name     = format("%s-ecommerce-fn-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host ecommerce function
+module "ecommerce_function_snet" {
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v4.3.2"
+  name                                           = format("%s-ecommerce-fn-snet", local.project)
+  address_prefixes                               = [var.cidr_subnet_ecommerce_functions]
+  resource_group_name                            = azurerm_resource_group.ecommerce_functions_rg.name
+  virtual_network_name                           = data.azurerm_virtual_network.vnet.name
+  enforce_private_link_endpoint_network_policies = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+  ]
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "ecommerce_function_app" {
+  source = "git::https://github.com/pagopa/azurerm.git//function_app?ref=v4.3.2"
+
+  resource_group_name = azurerm_resource_group.ecommerce_functions_rg.name
+  name                = format("%s-fn", local.project)
+  location            = var.location
+  health_check_path   = "info"
+  subnet_id           = module.ecommerce_function_snet.id
+  runtime_version     = "~4"
+  os_type             = "linux"
+  linux_fx_version    = "Java|17"
+
+  always_on                                = var.ecommerce_function_always_on
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+
+  app_service_plan_name = format("%s-plan-fn", local.project)
+  app_service_plan_info = {
+    kind                         = var.ecommerce_functions_app_sku.kind
+    sku_tier                     = var.ecommerce_functions_app_sku.sku_tier
+    sku_size                     = var.ecommerce_functions_app_sku.sku_size
+    maximum_elastic_worker_count = 0
+  }
+
+  storage_account_name = replace(format("%s-sa-fn", local.project), "-", "")
```

Comment:
Potential issue: there's a 24 character limit on storage account names, and the generated name `pagopadweuecommercesafn` is already 23 characters long

Reply:
Ugly, but what about `ecommtx`?

Reply:
sounds good :-)

Reply:
Fixed in [979eb98](https://github.com/pagopa/pagopa-infra/pull/692/commits/979eb98f0a0e50a79c8c4ec24ffcb3d0649deead)